spec/unit/provider/package/rpm_spec.rb

```diff
@@ -123,6 +123,13 @@
         provider.upgrade_package("ImageMagick-c++", "6.5.4.7-7.el6_5")
       end
 
+      it "should run rpm -U --oldpackage with the package source to downgrade" do
+        allow(new_resource).to receive(:allow_downgrade).and_return(true)
```

Comment:
I used the same code with [yum_spec.rb](https://github.com/chef/chef/blob/68bab0b362c47e5a75492e57094f72ecee1171ae/spec/unit/provider/package/yum_spec.rb#L443), so let me keep as is if you are okay.